City-of-Helsinki/drupal-helfi-platform-config
# UHF-3596: Add new fields and update remote-video configuration

#  Remote-video paragraph changes [UHF-3596](https://helsinkisolutionoffice.atlassian.net/browse/UHF-3596)
Add title and description field for the remote-video paragraph and separate title and description under the video itself. Style the paragraph to have black background and make sure that if there is remote-video paragraph right after another remote-video paragraph without a title and description for the paragraph the paragraphs should be attached to each other.

## What was done
* Added update hook to helfi_content that adds the four new fields to remote video paragraph.
* Added another update hook to helfi_tpr_config for the sites that have helfi_tpr on. This adds remote-videos to upper-content area for the unit and service entities that didn't have the possibility before.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `composer install`
    * `make drush-cr && make drush-cim && make drush-cr && make drush-updb`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-3596_remote_video_paragraph_changes && composer require drupal/hdbt:dev-UHF-3596_remote_video_paragraph_changes --with-all-dependencies`
* Run `make drush-updb drush-cr`

## How to test
* Repeat the following steps for basic page (upper and lower content area), landing page, unit (upper and lower content area), service (upper and lower content area) and news item.
* Edit an existing entity and add remote-video paragraph. The paragraph should now have 5 fields and first two are the title and description for the paragraph. Then you have the field to add the video and under it there is title and description for the video. The description for the video should be compulsory.
* Fill in the fields and add a second remote-video paragraph under the first one with just the description for the video field filled.
* Add one more remote-video paragraph under the second one but this time add title or description (or both) for the paragraph.
* Save the entity.
* Go to view the entity and make sure that first there is two video paragraphs that are attached together and then one separate one. The layout of the paragraphs should look similar to this: https://share.goabstract.com/6ee3d494-c081-4da3-a3d0-961b3f9e56ea
* Check that the paragraphs work responsively. Repeat the steps above to the aforementioned entity types.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/256
* https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/100
* https://github.com/City-of-Helsinki/drupal-helfi-strategia/pull/224
* https://github.com/City-of-Helsinki/drupal-helfi-kymp/pull/318
* https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/321
* https://github.com/City-of-Helsinki/drupal-helfi-asuminen/pull/53
* https://github.com/City-of-Helsinki/drupal-helfi-tyo-yrittaminen/pull/58
